parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -142,7 +144,8 @@ public void add(T value) {
   public Metrics metrics() {
     Preconditions.checkState(closed, "Cannot return metrics for unclosed writer");
     if (writer != null) {
-      return ParquetUtil.footerMetrics(writer.getFooter(), model.metrics(), metricsConfig);
+      return ParquetMetrics.metrics(
+          schema, parquetSchema, metricsConfig, writer.getFooter(), model.metrics());
```

Comment:
Schema is currently required because there is no annotation for variants in the Parquet schema.